Add docs for `Tabs` (#282)
mckinsey/vizro: vizro-core/docs/pages/user_guides/tabs.md

# How to use tabs

This guide shows you how to use tabs to organize your containers into subsections inside the dashboard.

To add [`Tabs`][vizro.models.Tabs] to your page, do the following:

1. Insert the [`Tabs`][vizro.models.Tabs] into the `components` argument of the [`Page`][vizro.models.Page]
2. Insert your [`Containers`][vizro.models.Container] into the `components` argument of the [`Tabs`][vizro.models.Tabs]

> 
> 

Reviewer: ```suggestion

Note that by using the [`Tabs`][vizro.models.Tabs] model, the following holds true:
- [`Filters`][vizro.models.Filter] affect their targets independent of tabs, i.e. they filter even when the target is not visible
- the `title` of the [`Containers`][vizro.models.Container] inserted will become the title of the tab
- if you would like to add a title inside the tab, you can add a nested [`Container`][vizro.models.Container]
- it is possible to have multiple containers per [`Page`][vizro.models.Page]

```

Author: I like it! I've rephrased a bit and added some additional information:

```
By using the [`Tabs`][vizro.models.Tabs], the following applies:
- If targets are not specified, [`Filters`][vizro.models.Filter] affect all components on all tabs, i.e., they also filter the components on the non-opened tab
- The `title` of the outer [`Container`][vizro.models.Container] inserted into `Tabs.components` will be displayed as a tab label and the title will removed from the `Container`
- If you want to keep the `Container.title` inside the tab, you can apply custom CSS
- If you want to have a different `Container.title` inside the tab, you can add a nested [`Container`][vizro.models.Container]

```

Reviewer: Sounds perfect! The only thing I would maybe change a little is the first bullet. The point about filters is true, but explains rather our filtering behaviour. I think the more interesting bit is that hidden tabs are filtered.

I assume this happens not only if no targets are specified (what we say), but also when a "currently hidden" target is specified (what we somewhat don't say). So maybe we can emphasize that a little more 🤔 Wdyt?

Author: I've rephrased a bit

                            data_frame: gapminder_2007
                            title: Graph 1
                            x: continent
                            y: lifeExp
                            color: continent
                      - type: graph
                        figure:
                            _target_: box
                            data_frame: gapminder_2007
                            title: Graph 2
                            x: continent
                            y: lifeExp
                            color: continent
                  - title: Tab II
                    type: container
                    components:
                      - type: graph
                        figure:
                            _target_: scatter
                            data_frame: gapminder_2007
                            title: Graph 3
                            x: gdpPercap
                            y: lifeExp
                            size: pop
                            color: continent
              type: tabs
          controls:
              - type: filter
                column: continent
          title: Tabs
        ```

>     === "Result"

Reviewer: I realize we need gifs... a lot of the advanced functionality just doesn't come across anymore.. discussed this also with Adam. But for a separate PR

       [![Tabs]][Tabs]

>     [Tabs]: ../../assets/user_guides/components/tabs.png

Author: The example currently only includes **single tabs with two containers**.

Do we want to add:
- Two tabs with containers on one page?
- Single Tabs with nested containers (this one is relevant if we want to show how to add a different title to the container)?
-

Author: **Two tabs with container on one page:**
![Screenshot 2024-01-24 at 10 08 31](https://github.com/mckinsey/vizro/assets/90609403/8b56a837-03e0-4154-9278-ceec4cc71c1d)

Author: **Single tabs with nested containers:**
- Would remove the container below and only show the single tabs with two tab secitons

![Screenshot 2024-01-24 at 10 08 50](https://github.com/mckinsey/vizro/assets/90609403/f4dd3d7d-75b0-48c9-a342-5e9487d58378)

Reviewer 2: I would keep it even leaning by not adding most of those notes at all 😅 

* Multiple tabs on one page? ➡️ this could be a nice example but also fine without
* Single Tabs with nested containers (this one is relevant if we want to show how to add a different title to the container)? ➡️ I would just ignore this altogether for now

# How to use tabs

> Tabs organize and separate groups of related content in a dashboard, letting users navigate between different sections or views within a click.

Reviewer 2: ```suggestion
Tabs organize and separate groups of related content in a dashboard, letting users switch between different sections or views.
```

This guide shows you how to use tabs to organize your containers into subsections inside the dashboard.

> 

Reviewer 2: We should have something here that explains what I was trying to say in https://github.com/McK-Internal/vizro-internal/issues/292#issuecomment-1866153211 about how tabs are really just a way of stacking containers. Getting this mental model clear is very important because it will help people understand how simple it is to configure a dashboard and how much sense (hopefully) it makes, rather than them feeling like they're needing to learn more and more concepts every time they want to change layout slightly.

Reviewer 2: Likewise we should have something that links from the containers page to here with something like "an alternative way of displaying multiple containers is to put them inside `Tabs` blah blah". Don't worry if you feel like you're repeating the same point in multiple places. People learn by repetition, and it's useful for a user to end up on a single documentation page and be linked to other ones that will help them understand related topics easily. We need to make sure that our docs are a cohesive whole rather than just a set of unrelated pages.

Author: Let me know if the current changes convey this sufficiently, but I definitely get your point 👍 

https://github.com/mckinsey/vizro/pull/282/commits/a1483f7b355faab846c8766f6cc64d987a85c7b4

To add [`Tabs`][vizro.models.Tabs] to your page, do the following:

1. Insert the [`Tabs`][vizro.models.Tabs] into the `components` argument of the [`Page`][vizro.models.Page]

> 2. Insert your [`Containers`][vizro.models.Container] into the `components` argument of the [`Tabs`][vizro.models.Tabs]

Reviewer 2: ```suggestion
2. Insert your [`Containers`][vizro.models.Container] into the `tabs` argument of the [`Tabs`][vizro.models.Tabs]
```

Author: Good one! 👁️

> By using the [`Tabs`][vizro.models.Tabs], the following applies:

Reviewer 2: ```suggestion
By using [`Tabs`][vizro.models.Tabs], the following applies:
```

- [`Filters`][vizro.models.Filter] affect all components on all tabs (opened and closed) of the page if not specified otherwise inside `Filter.targets`

> - The `title` of the outer [`Container`][vizro.models.Container] inserted into `Tabs.components` will be displayed as a tab label, and the title will removed from the `Container`

Reviewer 2: ```suggestion
- The `title` of the [`Container`][vizro.models.Container] inserted into `Tabs.tabs` will be displayed as a tab label, and the title will removed from the `Container`
```

- If you want to keep the `Container.title` inside the Tab, you can apply custom CSS

> - If you want a different `Container.title` inside the tab, add a nested [`Container`][vizro.models.Container]

Reviewer 2: I would actually even just remove these two points because "you can apply custom CSS" isn't super helpful if we don't specify what that CSS is, which means adding more detail than most people will care about, and the nested `Container` point still sounds like an unusual requirement to me and isn't fully explained here either. I would rather wait and see who (if anyone) asks about these two points. And if they do add a "proper" section on them.

If we really want them here already though then can we break the list up into two separate things instead of having the `Filter` point in the same list as the `title` ones? e.g. put the `Filter` thing in a separate note and then just have this list consist of the 3 points about title.

Author: No, I agree - let's just remove the two points and wait if people do request it 👍

# How to use tabs

Tabs organize and separate groups of related content in a dashboard, letting users switch between different sections or views.

> They are essentially a way of stacking multiple [`Containers`](container.md) on top of each other, and letting the user switch between them.

Reviewer 2: ```suggestion
They are essentially a way of putting multiple [`Containers`](container.md) in the same screen space, and letting the user switch between them.
```

[`Containers`][vizro.models.Container] enable the grouping of page components into sections and subsections.

> Both `Tabs` and `Containers` can be considered as a means to customize your page layout, and are an alternative to using [`Layout`](layout.md).
> However, if you simply want to arrange components on a page, we recommend reading our user guide on [`Layout`](layout.md) first.

Reviewer 2: ```suggestion
Both `Tabs` and `Containers` are a more advanced technique for customizing your page layout. If you simply want to arrange components on a page, we recommend reading our user guide on [`Layout`](layout.md) first.
```

# How to use tabs

[`Tabs`][vizro.models.Tabs]  organize and separate groups of related content in a dashboard, letting users switch between different sections or views.
They are essentially a way of putting multiple [`Containers`][vizro.models.Container] in the same screen space, and letting the user switch between them.

> `Containers` enable the grouping of page components into sections and subsections, see this user guide [here](container.md) for more information.

Reviewer 2: ```suggestion
`Containers` enable the grouping of page components into sections and subsections; see our [user guide on `Containers`](container.md) for more information.
```

Reviewer 2: See e.g. https://github.com/kedro-org/kedro/wiki/Kedro-documentation-style-guide#links 🙂

Author: Ah nice, I didn't even know that wiki existed. Super helpful! 🚀

Author: I've fixed it in other places as well where we have this pattern of "see user guide on.."
